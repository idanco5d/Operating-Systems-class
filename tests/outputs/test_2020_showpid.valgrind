==558== Memcheck, a memory error detector
==558== Copyright (C) 2002-2022, and GNU GPL'd, by Julian Seward et al.
==558== Using Valgrind-3.20.0 and LibVEX; rerun with -h for copyright info
==558== Command: /mnt/c/Users/illus/os/hw/1/smash
==558== Parent PID: 533
==558== 
==558== 
==558== FILE DESCRIPTORS: 8 open (3 std) at exit.
==558== Open file descriptor 11: /mnt/c/Users/illus/os/hw/1/tests/outputs/test_2020_showpid.out
==558==    at 0x4CB18CB: dup (syscall-template.S:120)
==558==    by 0x404FB6: SmallShell::handlePipeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) (in /mnt/c/Users/illus/os/hw/1/smash)
==558==    by 0x405570: SmallShell::handlePipeIoCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) (in /mnt/c/Users/illus/os/hw/1/smash)
==558==    by 0x405868: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (in /mnt/c/Users/illus/os/hw/1/smash)
==558==    by 0x4196AF: main (in /mnt/c/Users/illus/os/hw/1/smash)
==558== 
==558== Open file descriptor 8:
==558==    at 0x4CB18CB: dup (syscall-template.S:120)
==558==    by 0x404F7C: SmallShell::handlePipeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) (in /mnt/c/Users/illus/os/hw/1/smash)
==558==    by 0x405570: SmallShell::handlePipeIoCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) (in /mnt/c/Users/illus/os/hw/1/smash)
==558==    by 0x405868: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (in /mnt/c/Users/illus/os/hw/1/smash)
==558==    by 0x4196AF: main (in /mnt/c/Users/illus/os/hw/1/smash)
==558== 
==558== Open file descriptor 7: /mnt/c/Users/illus/os/hw/1/tests/outputs/test_2020_showpid.out
==558==    at 0x4CB18CB: dup (syscall-template.S:120)
==558==    by 0x404FB6: SmallShell::handlePipeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) (in /mnt/c/Users/illus/os/hw/1/smash)
==558==    by 0x405570: SmallShell::handlePipeIoCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) (in /mnt/c/Users/illus/os/hw/1/smash)
==558==    by 0x405868: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (in /mnt/c/Users/illus/os/hw/1/smash)
==558==    by 0x4196AF: main (in /mnt/c/Users/illus/os/hw/1/smash)
==558== 
==558== Open file descriptor 6:
==558==    at 0x4CB18CB: dup (syscall-template.S:120)
==558==    by 0x404F7C: SmallShell::handlePipeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) (in /mnt/c/Users/illus/os/hw/1/smash)
==558==    by 0x405570: SmallShell::handlePipeIoCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, int, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >, bool) (in /mnt/c/Users/illus/os/hw/1/smash)
==558==    by 0x405868: SmallShell::executeCommand(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >) (in /mnt/c/Users/illus/os/hw/1/smash)
==558==    by 0x4196AF: main (in /mnt/c/Users/illus/os/hw/1/smash)
==558== 
==558== Open file descriptor 3: /mnt/c/Users/illus/os/hw/1/tests/outputs/test_2020_showpid.valgrind
==558==    <inherited from parent>
==558== 
==558== 
==558== HEAP SUMMARY:
==558==     in use at exit: 0 bytes in 0 blocks
==558==   total heap usage: 97 allocs, 97 frees, 82,509 bytes allocated
==558== 
==558== All heap blocks were freed -- no leaks are possible
==558== 
==558== For lists of detected and suppressed errors, rerun with: -s
==558== ERROR SUMMARY: 0 errors from 0 contexts (suppressed: 0 from 0)
