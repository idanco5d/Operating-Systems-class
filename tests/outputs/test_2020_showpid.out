smash> smash pid is 8847
smash> smash pid is 8847
smash> smash> smash pid is 8847
smash> smash pid is 8847
smash> smash pid is 8847
smash> smash> 